lib/collection/src/operations/payload_ops.rs

```diff
@@ -1,58 +1,18 @@
-
-
 use serde;
 use serde::{Deserialize, Serialize};
 use schemars::{JsonSchema};
-use segment::types::{PointIdType, PayloadKeyType, PayloadType, GeoPoint};
+use segment::types::{PointIdType, PayloadKeyType, PayloadInterfaceStrict};
 use std::collections::HashMap;
 
 
-#[derive(Debug, Deserialize, Serialize, JsonSchema)]
-#[serde(rename_all = "snake_case")]
-#[serde(untagged)]
-pub enum PayloadVariant<T> {
-    Value(T),
-    List(Vec<T>)
-}
-
-impl<T: Clone> PayloadVariant<T> {
-    pub fn to_list(&self) -> Vec<T>{
-        match self {
-            PayloadVariant::Value(x) => vec![x.clone()],
-            PayloadVariant::List(vec) => vec.clone(),
-        }
-    }
-}
-
-#[derive(Debug, Deserialize, Serialize, JsonSchema)]
-#[serde(rename_all = "snake_case")]
-#[serde(tag = "type",  content = "value")]
-pub enum PayloadInterface {
-    Keyword(PayloadVariant<String>),
-    Integer(PayloadVariant<i64>),
-    Float(PayloadVariant<f64>),
-    Geo(PayloadVariant<GeoPoint>),
-}
-
-impl PayloadInterface {
-    pub fn to_payload(&self) -> PayloadType {
-        match self {
-            PayloadInterface::Keyword(x) => PayloadType::Keyword(x.to_list()),
-            PayloadInterface::Integer(x) => PayloadType::Integer(x.to_list()),
-            PayloadInterface::Float(x) =>  PayloadType::Float(x.to_list()),
-            PayloadInterface::Geo(x) => PayloadType::Geo(x.to_list()),
-        }
-    }
-}
-
 
 /// Define operations description for point payloads manipulation
 #[derive(Debug, Deserialize, Serialize, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum PayloadOps {
     /// Set payload value, overrides if it is already exists
     SetPayload {
-        payload: HashMap<PayloadKeyType, PayloadInterface>,
+        payload: HashMap<PayloadKeyType, PayloadInterfaceStrict>,
```

Comment:
`PayloadInterface` could be kept as is. So new options will also be available for REST interface.

Reply:
will change, just being conservative

Reply:
done now!